graspy/cluster/autogmm.py

```diff
@@ -0,0 +1,577 @@
+        If a list/array, it must be a list/array of strings containing only
+            'ward', 'complete', 'average', and/or 'single'.
+        
+    covariance_type : {'full', 'tied', 'diag', 'spherical', 'all' (default)} , optional
+        String or list/array describing the type of covariance parameters to use.
+        If a string, it must be one of:
+        
+        - 'full'
+            each component has its own general covariance matrix
+        - 'tied'
+            all components share the same general covariance matrix
+        - 'diag'
+            each component has its own diagonal covariance matrix
+        - 'spherical'
+            each component has its own single variance
+        - 'all'
+            considers all covariance structures in ['spherical', 'diag', 'tied', 'full']
+        If a list/array, it must be a list/array of strings containing only
+            'spherical', 'tied', 'diag', and/or 'spherical'.
+    
+    random_state : int, RandomState instance or None, optional (default=None)
+        If int, random_state is the seed used by the random number generator;
+        If RandomState instance, random_state is the random number generator;
+        If None, the random number generator is the RandomState instance used
+        by ``np.random``.
+
+    label_init : array-like, shape (n_samples,), optional (default=None)
+        List of labels for samples if available. Used to initialize the model.
+
+    max_iter : int, defaults to 100.
+        The maximum number of EM iterations to perform.
+
+    verbose : int, default to 0.
+        Enable verbose output. If 1 then it prints the current initialization and each iteration step.
+        If greater than 1 then it prints also the log probability and the time needed for each step.
+
+
+    Attributes
+    ----------
+    results_ : pandas.DataFrame
+        Contains exhaustive information about all the clustering runs.
+        Columns are:
+            'model' - fit GaussianMixture object
+            'bic' - Bayesian Information Criterion
+            'ari' - Adjusted Rand Index, nan if y is not given
+            'n_components' - number of clusters
+            'affinity' - affinity used in Agglomerative Clustering
+            'linkage' - linkage used in Agglomerative Clustering
+            'covariance_type' - covariance type used in GMM
+            'reg_covar' : regularization used in GMM
+
+    bic_ : the best (lowest) Bayesian Information Criterion
+    n_components_ : number of clusters in the model with the best BIC
+    covariance_type_ : covariance type in the model with the best BIC
+    affinity_ : affinity used in the model with the best BIC
+    linkage_ : linkage used in the model with the best BIC
+    reg_covar_ : regularization used in the model with the best BIC
+    ari_ : ARI from the model with the best BIC, nan if no y is given
+    model_ : GaussianMixture object with the best BIC
```

Comment:
is there a citation we can add for all of this, maybe one or two of the mclust papers?